File: CS-Examples/09_Charts/GetWorksheetOfChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts
Imports System.Text

Namespace GetWorksheetOfChart
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object.
            Dim workbook As New Workbook()

            ' Load the Excel file into the workbook.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartToImage.xlsx")

            ' Get the first worksheet in the workbook.
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Get the first chart in the worksheet.
            Dim chart As Chart = worksheet.Charts(0)

            ' Cast the chart's worksheet as a Worksheet object.
            Dim wSheet As Worksheet = TryCast(chart.Worksheet, Worksheet)

            ' Create a StringBuilder object to store the content.
            Dim content As New StringBuilder()

            ' Create a string with the sheet name and the chart's sheet name.
            Dim result As String = String.Format("Sheet Name: " & worksheet.Name & vbCrLf & "Charts' sheet Name: " & wSheet.Name)

            ' Append the result string to the content StringBuilder.
            content.AppendLine(result)

            ' Specify the output file name.
            Dim outputFile As String = "Output.txt"

            ' Write the contents of the content StringBuilder to a text file.
            File.WriteAllText(outputFile, content.ToString())
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/09_Charts/GetWorksheetOfChart.cs
using System;
using System.Windows.Forms;
using System.IO;
using Spire.Xls;
using System.Text;


namespace GetWorksheetOfChart
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartToImage.xlsx");

            //Access first worksheet of the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            //Access the first chart inside this worksheet
            Chart chart = worksheet.Charts[0];

            //Get its worksheet
            Worksheet wSheet = chart.Worksheet as Worksheet;

            //Create StringBuilder to save 
            StringBuilder content = new StringBuilder();

            //Set string format for displaying
            string result = string.Format("Sheet Name: " + worksheet.Name + "\r\nCharts' sheet Name: " + wSheet.Name);

            //Add result string to StringBuilder
            content.AppendLine(result);

            //String for output file 
            String outputFile = "Output.txt";

            //Save them to a txt file
            File.WriteAllText(outputFile, content.ToString());

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the output file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
